scripts/distribute_bpe_reward.py

```diff
@@ -0,0 +1,86 @@
+# coding: utf-8
+
+"""
+To distribute token-level rewards to sub-word units.
+"""
+
+import argparse
+import re
+import sys
+
+import numpy as np
+
+
+def main(args):
+
+    target_file = args.target
+    reward_file = args.reward
+
+    with open(target_file, 'r') as bpe_in, \
+         open(reward_file, 'r') as fbkin, open('feedback.bpe', 'w') as fbkout:
+
+        trans = bpe_in.readlines()
+        rewards = fbkin.readlines()
+
+        for idx, (sent, reward) in enumerate(zip(trans, rewards)):
+            reward_tok = reward.split()
+
+            if '@@' not in sent:
+                # No sub-words in this sentence
+                bpe_reward = reward_tok
+            else:
+                sent_str = sent.replace("@@ ", "")
+                sent_tok = sent.split()
+                assert(len(sent_str.split())==len(reward_tok)), \
+                    "input sentence length should be equal to number of reward"
+
+                if idx % args.print_interval == 0:
+                    print("sentence (post-process bpe):{}".format(
+                          sent_str[:-1]))
+                    print('input reward:{}'.format(" ".join(reward_tok)))
+
+                loc_bpe = [loc for loc, tok in enumerate(sent_tok) 
+                            if "@@" in tok]
+                loc_bpe_in_original = [bpeloc - len(loc_bpe[:idx]) 
+                                        for idx, bpeloc in enumerate(loc_bpe)]
```

Comment:
shouldn't len(loc_bpe[:idx]) == idx+1?
I'm not sure I understand what this line is doing but it looks like it's too complicated ;)

Reply:
I have added a illustrative example, and re-name some variables. It should be more self-explanatory now. Hopefully :) 

Basically, what I did is to map the bpe split token, e,g. "To@@ ", "day", back to its unsplit format, "Today" so that I can assign the reward to "To@@", and "day" by using the reward for "Today".